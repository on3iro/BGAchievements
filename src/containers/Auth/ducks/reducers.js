import { combineReducers } from 'redux'

import * as types from './actionTypes'

export const user = (state = {
  isLoggedIn: false,
  hasRegistered: false,
  token: '',
  data: {},
  error: {}
}, action) => {
  switch (action.type) {
    case types.LOGIN_SUCCESS: {
      const data = action.payload

      return {
        ...state,
        isLoggedIn: true,
        token: data.token,
        data: data.user,
        error: ''
      }
    }

    case types.REGISTER_SUCCESS: {
      return {
        ...state,
        hasRegistered: true
      }
    }

    case types.REGISTER_ERROR:
    case types.LOGIN_ERROR: {
      const error = action.payload

      return {
        ...state,
        error: error
      }
    }

    case types.LOGOUT: {
      return {
        ...state,
        isLoggedIn: false,
        token: '',
        data: {}
      }
    }

    case types.UPDATE_MAX_ACCOUNTS: {
      return {
        ...state,
        data: {
          ...state.data,
          maxAccounts: action.payload
        }
      }
    }

    default: {
      return state
    }
  }
}

export const AuthRequestState = (state = {
  isLoading: false
}, action) => {
  switch (action.type) {
    case types.LOGIN_SUBMIT: {
      return { ...state, isLoading: true }
    }

    case types.REGISTER_SUBMIT: {
      return { ...state, isLoading: true }
    }

    case types.LOGIN_SUCCESS: {
      return { ...state, isLoading: false }
    }

    case types.REGISTER_SUCCESS: {
      return { ...state, isLoading: false }
    }

    case types.REGISTER_ERROR: {
      return { ...state, isLoading: false }
    }

    case types.LOGIN_ERROR: {
      return { ...state, isLoading: false }
    }

    default:
      return state
  }
}

const authReducer = combineReducers({
  user,
  AuthRequestState
})

export default authReducer
