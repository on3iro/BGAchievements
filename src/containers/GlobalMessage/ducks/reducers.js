import { combineReducers } from 'redux'

import * as types from './actionTypes'

export const GlobalError = (state = '', action) => {
  switch (action.type) {
    case types.GLOBAL_ERROR: {
      return action.payload
    }

    case types.CLEAR_MESSAGE: {
      return ''
    }

    default: {
      return state
    }
  }
}

export const GlobalMessage = (state = '', action) => {
  switch (action.type) {
    case types.GLOBAL_MESSAGE: {
      return action.payload
    }

    case types.CLEAR_MESSAGE: {
      return ''
    }

    default: {
      return state
    }
  }
}

const GlobalMessages = combineReducers({
  GlobalError,
  GlobalMessage
})

export default GlobalMessages
