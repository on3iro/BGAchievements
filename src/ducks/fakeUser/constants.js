// @flow

export const INIT: 'INIT' = 'INIT'
export const USER_FETCH_SUCCESS: 'USER_FETCH_SUCCESS' = 'USER_FETCH_SUCCESS'
export const USER_FETCH_FAIL: 'USER_FETCH_FAIL' = 'USER_FETCH_FAIL'
