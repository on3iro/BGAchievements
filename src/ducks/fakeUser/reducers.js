// @flow

import { loop, Cmd } from 'redux-loop'

import {
  INIT,
  USER_FETCH_FAIL,
  USER_FETCH_SUCCESS
} from './constants'

import type {
  Action,
  State
} from './types'

import {
  fetchUser,
  userFetchFailedAction,
  userFetchSuccessfulAction
} from './actions'

const fakeUserReducer = (state: State = {
  loading: false,
  user: null,
  error: null
}, action: Action): State => {
  switch (action.type) {
    case INIT:
      return loop(
        {...state, loading: true},
        Cmd.run(fetchUser, {
          successActionCreator: userFetchSuccessfulAction,
          failActionCreator: userFetchFailedAction,
          args: ['userA']
        })
      )

    case USER_FETCH_SUCCESS:
      return { ...state, user: action.payload, loading: false }

    case USER_FETCH_FAIL:
      return { ...state, error: action.payload, loading: false }

    default:
      return state
  }
}

export {
  fakeUserReducer
}
