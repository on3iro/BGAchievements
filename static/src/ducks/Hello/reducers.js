import { combineReducers, loop, Cmd } from 'redux-loop'

import * as actions from './actions'
import sideeffects from './sideeffects'

const HelloReducer = (state = [], action) => {
  switch (action.type) {
    case 'SAMPLE_DATA_FETCH_START': {
      return loop(state, Cmd.run(sideeffects.fetchSampleData, {
        successActionCreator: actions.fetchSampleDataSuccess,
        errorActionCreator: actions.fetchSampleDataFailure
      }))
    }

    case 'SAMPLE_DATA_FETCH_SUCCESS': {
      return action.payload
    }

    case 'SAMPLE_DATA_FETCH_FAILURE': {
      return []
    }

    default: {
      return state
    }
  }
}

const LoadingReducer = (state = false, action) => {
  switch (action.type) {
    case 'SAMPLE_DATA_FETCH_START': {
      return true
    }

    case 'SAMPLE_DATA_FETCH_SUCCESS':
    case 'SAMPLE_DATA_FETCH_FAILURE': {
      return false
    }

    default: {
      return state
    }
  }
}

export default combineReducers({
  sampleData: HelloReducer,
  loading: LoadingReducer
})
