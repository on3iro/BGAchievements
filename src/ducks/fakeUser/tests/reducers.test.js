import * as reducers from '../reducers'
import * as constants from '../constants'

describe('FakeUser reducers', () => {
  describe('fakeUserReducer', () => {
    it('Should handle INIT', () => {
      expect(reducers.fakeUserReducer(undefined, {
        type: constants.INIT
      })).toMatchSnapshot()
    })

    it('Should handle USER_FETCH_SUCCESS', () => {
      expect(reducers.fakeUserReducer(undefined, {
        type: constants.USER_FETCH_SUCCESS,
        payload: 'dummyUser'
      })).toMatchSnapshot()
    })

    it('Should handle USER_FETCH_FAIL', () => {
      expect(reducers.fakeUserReducer(undefined, {
        type: constants.USER_FETCH_FAIL,
        payload: 'dummyError'
      })).toMatchSnapshot()
    })
  }) // end fakeUserReducer
})
