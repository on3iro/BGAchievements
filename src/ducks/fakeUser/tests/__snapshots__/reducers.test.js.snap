// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FakeUser reducers fakeUserReducer Should handle INIT 1`] = `
Array [
  Object {
    "error": null,
    "loading": true,
    "user": null,
  },
  Object {
    "args": Array [
      "userA",
    ],
    "failActionCreator": [Function],
    "func": [Function],
    "simulate": [Function],
    "successActionCreator": [Function],
    "type": "RUN",
    Symbol(isCmd): true,
  },
]
`;

exports[`FakeUser reducers fakeUserReducer Should handle USER_FETCH_FAIL 1`] = `
Object {
  "error": "dummyError",
  "loading": false,
  "user": null,
}
`;

exports[`FakeUser reducers fakeUserReducer Should handle USER_FETCH_SUCCESS 1`] = `
Object {
  "error": null,
  "loading": false,
  "user": "dummyUser",
}
`;
